lib/screens/surveydetails/widget/answer_dropdown_widget.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+import 'package:lydiaryanfluttersurvey/model/ui/question_ui_model.dart';
+import 'package:lydiaryanfluttersurvey/resources/dimensions.dart';
+import 'package:scroll_snap_list/scroll_snap_list.dart';
+
+const int _dropdownItemLength = 3;
+const double _dropdownThreshold = 20.0;
+
+class AnswerDropdownWidget extends StatefulWidget {
+  final QuestionUiModel question;
+  final Function(String) onSelect;
+
+  const AnswerDropdownWidget({
+    Key? key,
+    required this.question,
+    required this.onSelect,
+  }) : super(key: key);
+
+  @override
+  State<AnswerDropdownWidget> createState() => _AnswerDropdownWidgetState();
+}
+
+class _AnswerDropdownWidgetState extends State<AnswerDropdownWidget> {
+  int _focusedIndex = 0;
+
+  Color? _getAnswerColor(int index) =>
+      _focusedIndex == index ? Colors.white : Colors.white54;
+
+  FontWeight _getAnswerFontWeight(int index) =>
+      _focusedIndex == index ? FontWeight.bold : FontWeight.normal;
+
+  void _onItemFocus(int index) {
+    setState(() {
+      _focusedIndex = index;
+    });
+    widget.onSelect(widget.question.answers[index].text ?? "");
+  }
+
+  Widget _buildListItem(BuildContext context, int index) {
+    return Container(
+      height: Dimensions.answerDropdownHeight,
+      padding: const EdgeInsets.symmetric(
+        horizontal: Dimensions.paddingLarge,
+      ),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Padding(
+            padding: const EdgeInsets.symmetric(
+              vertical: Dimensions.paddingSmall,
+            ),
+            child: Container(
+              alignment: Alignment.center,
+              child: Text(widget.question.answers[index].text ?? "",
+                  style: Theme.of(context).textTheme.bodyLarge?.copyWith(
+                        color: _getAnswerColor(index),
+                        fontWeight: _getAnswerFontWeight(index),
+                      )),
+            ),
+          ),
+          index < widget.question.answers.length - 1
+              ? const Divider(
+                  color: Colors.white,
+                  thickness: Dimensions.dividerWidth,
+                )
+              : const SizedBox(),
+        ],
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: SizedBox(
+        height: (_dropdownItemLength * Dimensions.answerDropdownHeight) -
+            _dropdownThreshold,
+        child: Column(
+          children: <Widget>[
+            Expanded(
+              child: ScrollSnapList(
+                scrollDirection: Axis.vertical,
+                onItemFocus: _onItemFocus,
+                itemSize: Dimensions.answerDropdownHeight,
+                itemBuilder: _buildListItem,
+                itemCount: widget.question.answers.length,
+              ),
+            ),
+          ],
+        ),
```

Comment:
@lydiasama do we need `Column` in this widget? It has only one child, so I think we can optimize to:
```suggestion
      child: SizedBox(
        height: (_dropdownItemLength * Dimensions.answerDropdownHeight) -
            _dropdownThreshold,
        child: ScrollSnapList(
                scrollDirection: Axis.vertical,
                onItemFocus: _onItemFocus,
                itemSize: Dimensions.answerDropdownHeight,
                itemBuilder: _buildListItem,
                itemCount: widget.question.answers.length,
              ),
```

Reply:
@manh-t I'm not sure if it affects to the `ScrollSnapList`. Let me try. 🙏🏻

Reply:
@manh-t It works! 👍🏻  Actually, I tried but it didn't work for some reason. Maybe after I refactored the UI could help. BTW, thank you for spotting on this.

Reply:
Fixed in 1406edd4